perfdash/config.go

```diff
@@ -389,48 +389,60 @@ func getProwConfig(configPaths []string) (Jobs, error) {
 			return nil, fmt.Errorf("error unmarshaling prow config from %s: %v", configPath, err)
 		}
 		for _, periodic := range conf.Periodics {
-			var thisPeriodicConfig Tests
-			for _, tag := range periodic.Tags {
-				if strings.HasPrefix(tag, "perfDashPrefix:") {
-					split := strings.SplitN(tag, ":", 2)
-					thisPeriodicConfig.Prefix = strings.TrimSpace(split[1])
-					continue
-				}
-				if strings.HasPrefix(tag, "perfDashJobType:") {
-					split := strings.SplitN(tag, ":", 2)
-					jobType := strings.TrimSpace(split[1])
-					var exists bool
-					if thisPeriodicConfig.Descriptions, exists = jobTypeToDescriptions[jobType]; !exists {
-						fmt.Fprintf(os.Stderr, "warning: unknown job type - %s\n", jobType)
-					}
-					continue
-				}
-				if strings.HasPrefix(tag, "perfDashBuildsCount:") {
-					split := strings.SplitN(tag, ":", 2)
-					i, err := strconv.Atoi(strings.TrimSpace(split[1]))
-					if err != nil {
-						fmt.Fprintf(os.Stderr, "warning: unparsable builds count - %v\n", split[1])
-					}
-					if i < 1 {
-						fmt.Fprintf(os.Stderr, "warning: non-positive builds count - %v\n", i)
-						continue
-					}
-					thisPeriodicConfig.BuildsCount = i
-					continue
-				}
-				if strings.HasPrefix(tag, "perfDash") {
-					fmt.Fprintf(os.Stderr, "warning: unknown perfdash tag name: %q\n", tag)
-				}
+			 config, err := parsePeriodicConfig(periodic)
+			 if err != nil {
+				fmt.Fprintf(os.Stderr, "warning: failed to parse config of %q due to: %v\n",
+					periodic.Name, err)
+				continue
 			}
-			if thisPeriodicConfig.Prefix == "" && thisPeriodicConfig.Descriptions == nil {
+			if config.Prefix == "" && config.Descriptions == nil {
```

Comment:
While you are modifying this, can we move this to parsePeriodicConfig as well?

I was thinking about something like:
config, hasConfig, err := parsePeridicConfig(periodic)
if err != nil {
  // log error
}
if hasConfig {
  jobs[periodic.Name] = config
}

if hasConfig is true if config.Prefix != "" or config.Descriptions != "".

Other way to do that is to say that parsePeriodicConfig can return nil instead of hasConfig.

WDYT?

Reply:
discussed offline, extracted validation logic to yet another method